MericCelik/assignment3/MainTest.java

```diff
@@ -0,0 +1,69 @@
+import org.junit.Test;
+import org.junit.Assert;
+import java.util.Arrays;
+import java.util.HashSet;
+
+/**@author AycaMericCelik
+ * */
+public class MainTest {
```

Comment:
Very nice tests! :) How would you feel about adding some more unit tests for other methods (like isWord and isPrefix in both of your dictionary implementations) or a tests with your own grid and dictionary where you test for different edge-cases (like illegal words with re-using letters, illegal words that would work if jumping over cells was allowed, words on diagonals, reverse diagonals etc.)

Reply:
Thank you so much! I added some more :blush:

Reply:
Oh wow, amazing job! So many tests! :D